### PR TITLE
Formatting and tidying

### PR DESCRIPTION
**Pull request by harmishhk**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
```
Comment (hogabrie) [inline .editorconfig]: We should no commit editor related configs.

File: include/anymal_teb_local_planner/equivalence_relations.h [hunk 2]
```diff
@@ -60,44 +59,39 @@ namespace anymal_teb_local_planner
...
+  virtual bool isValid() const = 0;
 
+  /**
+   * @brief Check if the trajectory is non-looping around an obstacle
+   * @return Returns false, if the trajectory loops around an obstacle
+   */
+  virtual bool isReasonable() const = 0;
 };
 
 using EquivalenceClassPtr = boost::shared_ptr<EquivalenceClass>;
```
Comment (hogabrie) [inline include/anymal_teb_local_planner/equivalence_relations.h]: std::shared_ptr?

**Comment by harmishhk** [comment]
It contains only formatting and tidying related changes. Tested to work in simulation.

File: .editorconfig (hunk 1 above)
Comment (harmishhk) [reply .editorconfig]: I thought it is useful as it works with many editors. https://editorconfig.org/ Anyway, then I will remove.

File: include/anymal_teb_local_planner/equivalence_relations.h (hunk 2 above)
Comment (harmishhk) [reply include/anymal_teb_local_planner/equivalence_relations.h]: That will be a logical/code change. For later...

File: .editorconfig (hunk 1 above)
Comment (hogabrie) [reply .editorconfig]: Hmm, you could add it to the gitignore then you can still have it locally :smile:

File: CMakeLists.txt [hunk 3]
```diff
@@ -258,6 +253,25 @@ install(DIRECTORY
   PATTERN ".svn" EXCLUDE
 )
 
+#################
+## Clang Tools ##
+#################
+# Generate compile_commands.json (needed if clang-tidy is run)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (remod) [inline CMakeLists.txt]: Isn't it too late to set this here?

Comment (harmishhk) [reply CMakeLists.txt]: It had no effect at compile time, as far as I can see, but I anyway moved it before the Build section.

Comment (remod) [reply CMakeLists.txt]: Yes, but you probably need to run it twice such that the compile commands are there for clang

**Comment by harmishhk** [comment]
@hogabrie By mistake I dismissed your last review. Could you please approve again, so I can go ahead with merge, sorry.

**Pull request merged**